src/deviceCommunication/commUSB.cpp

```diff
@@ -52,22 +52,21 @@ void CommUSB::sendData(const QByteArray& rawData) {
 };
 
 void CommUSB::readData() {
-    /// @todo add proper parser
-    QByteArray data = serialPort.readAll();
-    COMbuffer += data;
-    QStringList COMBufferList = COMbuffer.split('\n');
-    for (int i = 0; i < COMBufferList.length(); ++i) {
-        QString currentMsg = COMBufferList[i];
-        if (currentMsg.endsWith('\r')) {
-            float force = currentMsg.mid(1, 6).remove('-').toFloat();
-
-            COMbuffer.remove(currentMsg);
-            emit newForceDevice(currTime, force);
-            
-            currTime += 1.0f/40.0f; // TODO: use actual sampling rate
+    COMbuffer += serialPort.readAll();
```

Comment:
Use fixed size (e.g. `2*PACKET_EXPECTED_LEN`) `COMbuffer` and read the data with `QSerialPort::readData(char *data, qint64 maxSize)`. This method also reports errors which we can use to automatically disconnect and report to the user.

Reply:
Prever to stick with the `readAll()`; to limit the number of (slow) accesses to the serialPort
I tried `readData()` with `2*PACKET_EXPECTED_LEN`; however this resulted in a huge performance drop at 1280Hz

Reply:
I thought that would increase performane, but apparently not.